datafusion/functions/src/datetime/to_local_time.rs

```diff
@@ -0,0 +1,601 @@
+
+        let res = ToLocalTimeFunc::new().invoke(&[ColumnarValue::Scalar(
+            ScalarValue::TimestampNanosecond(Some(1), Some("Europe/Brussels".into())),
+        )]);
+        assert!(res.is_ok());
+
+        //
+        // Fallible test cases
+        //
+
+        // invalid number of arguments -- no argument
+        let res = ToLocalTimeFunc::new().invoke(&[]);
+        assert_eq!(
+            res.err().unwrap().strip_backtrace(),
+            "Execution error: to_local_time function requires 1 argument, got 0"
+        );
+
+        // invalid number of arguments -- more than 1 argument
+        let res = ToLocalTimeFunc::new().invoke(&[
+            ColumnarValue::Scalar(ScalarValue::TimestampSecond(Some(1), None)),
+            ColumnarValue::Scalar(ScalarValue::Utf8(Some("some string".to_string()))),
+        ]);
+        assert_eq!(
+            res.err().unwrap().strip_backtrace(),
+            "Execution error: to_local_time function requires 1 argument, got 2"
+        );
+
+        // invalid argument data type
+        let res = ToLocalTimeFunc::new().invoke(&[ColumnarValue::Scalar(
+            ScalarValue::Utf8(Some("some string".to_string())),
+        )]);
+        assert_eq!(
+            res.err().unwrap().strip_backtrace(),
+            "Execution error: to_local_time function requires timestamp argument, got Utf8"
+        );
+
+        // TODO chunchun: invalid timestamp: us, ms, s
+    }
+
+    #[test]
+    fn test_to_local_time_scalar() {
+        let timezone = Some("Europe/Brussels".into());
+        let timestamps_with_timezone = vec![
+            (
+                ScalarValue::TimestampNanosecond(
+                    Some(1_123_123_000_000_000_000),
+                    timezone.clone(),
+                ),
+                ScalarValue::TimestampNanosecond(Some(1_123_130_200_000_000_000), None),
+            ),
+            (
+                ScalarValue::TimestampMicrosecond(
+                    Some(1_123_123_000_000_000),
+                    timezone.clone(),
+                ),
+                ScalarValue::TimestampMicrosecond(Some(1_123_130_200_000_000), None),
+            ),
+            (
+                ScalarValue::TimestampMillisecond(
+                    Some(1_123_123_000_000),
+                    timezone.clone(),
+                ),
+                ScalarValue::TimestampMillisecond(Some(1_123_130_200_000), None),
+            ),
+            (
+                ScalarValue::TimestampSecond(Some(1_123_123_000), timezone),
+                ScalarValue::TimestampSecond(Some(1_123_130_200), None),
+            ),
+        ];
+
+        for (input, expected) in timestamps_with_timezone {
+            test_to_local_time_helpfer(input, expected);
+        }
+    }
+
+    #[test]
+    fn test_timezone_with_daylight_savings() {
+        let timezone_str = "America/New_York";
+
+        let test_cases = vec![
+            // DST time
+            "2019-03-31T04:00:00",
```

Comment:
I recommend we also hard code the expected timestamp value here (and we can verify it against `timestamp_str.parse()`) -- otherwise this tests is effectively testing that the udf does the same thing as `chrono` rather than get the expected output.

Reply:
Updated in https://github.com/apache/datafusion/pull/11347/commits/db5b73ebb44ee80dcef938dd631395e69b03af1b